tests/saitests/sai_qos_tests.py

```diff
@@ -1210,18 +1228,18 @@ def runTest(self):
             if hwsku == 'DellEMC-Z9332f-M-O16C64' or hwsku == 'DellEMC-Z9332f-O32':
                 send_packet(
                     self, src_port_id, pkt2, 
-                    pkts_num_egr_mem + pkts_num_leak_out + margin + pkts_num_dismiss_pfc - 1 + hysteresis
+                    (pkts_num_egr_mem + pkts_num_leak_out + pkts_num_dismiss_pfc + hysteresis) // cell_occupancy + margin - 1
                 )
             elif 'cisco-8000' in asic_type:
                 assert(fill_leakout_plus_one(self, src_port_id, dst_port_2_id, pkt2, int(self.test_params['pg']), asic_type))
                 send_packet(
                     self, src_port_id, pkt2,
-                    pkts_num_leak_out + margin + pkts_num_dismiss_pfc - 2 + hysteresis
+                    (pkts_num_leak_out + pkts_num_dismiss_pfc + hysteresis) // cell_occupancy + margin - 2
                 )
             else:
                 send_packet(
-                    self, src_port_id, pkt2, 
-                    pkts_num_leak_out + margin + pkts_num_dismiss_pfc - 1 + hysteresis
+                    self, src_port_id, pkt2,
+                     (pkts_num_leak_out + pkts_num_dismiss_pfc  + hysteresis) // cell_occupancy + margin - 1
```

Comment:
Please remove the blank.

Reply:
@stephenxs, what is the reason for the failure with smaller packet sizes?

Reply:
@neethajohn it's not smaller packet sizes. it's very small packets, like 64.
basically, it's because a large number of the lossless congested packets (much more than 50%) are small packets like 64. in such a case, the packet can run out its proportion of shared buffer pool and trek into headroom in advance.

Reply:
@stephenxs , how do we make sure that accounting on 4600c works correctly for smaller packet sizes?